### ISSUE TITLE
ttnn.pad docs should be improved

### ISSUE THREAD

**Issue by mert-kurttutan** [mert-kurttutan/issue-buffer#29]
see e.g. below: it is not obvious whicch dims should be expand from the padding dims argumetn given: 

```python


#!/usr/bin/env python3

import traceback

import torch

import ttnn


def maybe_unwrap_pad_result(pad_result):
    # Some docs mention list[Tensor]; handle both tensor and list return styles.
    if isinstance(pad_result, list):
        if len(pad_result) != 1:
            raise RuntimeError(f"Expected a single tensor from ttnn.pad, got {len(pad_result)}")
        return pad_result[0]
    return pad_result


def run_pad(input_tensor: ttnn.Tensor, padding: list[tuple[int, int]], value: float) -> None:
    print(f"input_shape={input_tensor.shape}, padding={padding}, value={value}")
    padded = maybe_unwrap_pad_result(ttnn.pad(input_tensor, padding, value))
    print(f"output_shape={padded.shape}")

def run_device_case() -> None:
    torch.manual_seed(0)
    device = None
    try:
        device = ttnn.CreateDevice(device_id=0, l1_small_size=16384)
        x_torch = torch.randn(2, 13, 4, dtype=torch.bfloat16)
        x_tt = ttnn.from_torch(
            x_torch,
            dtype=ttnn.bfloat16,
            layout=ttnn.ROW_MAJOR_LAYOUT,
            device=device,
        )
        run_pad(x_tt, padding=[(0, 0), (0, 3), (0, 0)], value=0.0)

        x_tt_2 = ttnn.from_torch(
            x_torch,
            dtype=ttnn.bfloat16,
            layout=ttnn.ROW_MAJOR_LAYOUT,
            device=device,
        )
        run_pad(x_tt_2, padding=[(0, 0), (0, 3)], value=0.0)
    finally:
        if device is not None:
            ttnn.close_device(device)


def main() -> None:
    try:
        print("Device 3D pad case")
        run_device_case()
    except Exception:
        print("Caught exception:")
        traceback.print_exc()


if __name__ == "__main__":
    main()

```
